README.md

```diff
@@ -130,22 +130,36 @@ above, `sonatypeMavenCentral()` is also available, which transparently sets up
 the [nexus publish plugin](https://github.com/marcphilipp/nexus-publish-plugin)
 to automatically deploy to a staging repository.
 
+Note that you should use the `sonatypeSnapshots()` repository for publishing snapshots.
+
 <details>
 <summary>Example</summary>
 
 ```groovy
 publishing {
-    repositories {
-        sonatypeMavenCentral {
-            // If you need to publish to a different repository
-            url = "https://s01.oss.sonatype.org/service/local/"
-            // The default is https://oss.sonatype.org/service/local/
-            credentials {
-                username 'XYZ'
-                password 'some-password'
-            }
+  repositories {
+    // Switch which repository is used based on if the version is a snapshot
+    if("${project.version}".endsWith('-SNAPSHOT')) {
+      sonatypeSnapshots {
+        // The default is https://oss.sonatype.org/content/repositories/snapshots/
+        url = "https://s01.oss.sonatype.org/content/repositories/snapshots/"
+        credentials {
+          username 'XYZ'
+          password 'some-password'
         }
+      }
+    } else {
+      sonatypeMavenCentral {
+        // If you need to publish to a different repository
+        // The default is https://oss.sonatype.org/service/local/
+        url = "https://s01.oss.sonatype.org/service/local/"
+        credentials {
+          username 'XYZ'
+          password 'some-password'
+        }
+      }
```

Comment:
Is `sonatypeSnapshots` and `sonatypeMavenCentral` significant, if you configure the url anyway ?

Would something like this also work ?

```groovy
repositories {
    customSonatypeMavenRepo {
        url = "${project.version}".endsWith('-SNAPSHOT') ? 'https://s01.oss.sonatype.org/content/repositories/snapshots/' : 'https://s01.oss.sonatype.org/service/local/'
        credentials {
          username 'XYZ'
          password 'some-password'
        }
      }
}
```

Reply:
The distinction is very important, because `sonatypeMavenCentral()` is not just a maven repository, but it configures the nexus publishing plugin under the hood instead.

Changing that URL to a snapshots URL would result in the plugin attempting (and failing) to call the service API to provision a new staging repository.